File: 프로그래머스/3/43162. 네트워크/네트워크.py
import sys


def dfs(i, computers, visited):
    for j in range(len(computers)):
        if not visited[j] and computers[i][j] == 1:
            visited[j] = True
            dfs(j, computers, visited)


def solution(n, computers):
    visited = [False] * n
    cnt = 0
    for i in range(n):
        if not visited[i]:
            visited[i] = True
            cnt += 1
            dfs(i, computers, visited)

    return cnt

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let isVisit = Array.from({length : n}).fill(false);
    
    let cnt = 0;
    for(let i=0 ; i < n ; i++){
        if (isVisit[i]) continue;
        cnt += 1;
        dfs(i,isVisit,computers, n);
    }
    
    return cnt;
}

let dfs = (i, isVisit,computers, n) => {
    for(let a = 0; a < n; a++){
        if(computers[i][a] === 1 && !isVisit[a]) {
            isVisit[a] = true;
            dfs(a,isVisit,computers, n);
        }
    }
}